app/src/main/java/com/x8bit/bitwarden/data/vault/manager/VaultLockManagerImpl.kt

```diff
@@ -227,7 +227,8 @@ class VaultLockManagerImpl(
         )
 
         if (invalidUnlockAttempts >= MAXIMUM_INVALID_UNLOCK_ATTEMPTS) {
-            userLogoutManager.logout(userId)
+            setVaultToLocked(userId = userId)
+            userLogoutManager.logout(userId = userId)
```

Comment:
Is it possible for the `VaultLockManager` to just observe accounts and whenever there is one that logs out in some way, we automatically lock?

Reply:
Yeah I was thinking that, would be easy to do with adding VLM to the ULM as a dependency.

Reply:
The `UserLogoutManager` is already injected into the `VaultLockManager`, you can just observe the login state from the `UserLogoutManager` there.